src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -2816,8 +2816,23 @@ address MacroAssembler::trampoline_call(Address entry, CodeBuffer* cbuf) {
          entry.rspec().type() == relocInfo::static_call_type ||
          entry.rspec().type() == relocInfo::virtual_call_type, "wrong reloc type");
 
+  bool need_trampoline = far_branches();
+  if (!need_trampoline && entry.rspec().type() == relocInfo::runtime_call_type && !CodeCache::contains(entry.target())) {
+    // If it is a runtime call of an address outside small CodeCache,
+    // we need to check whether it is in range.
+    address target = entry.target();
+    assert(target < CodeCache::low_bound() || target >= CodeCache::high_bound(), "target is inside CodeCache");
+    // Case 1: -------T-------L====CodeCache====H-------
+    //                ^-------longest branch---|
+    // Case 2: -------L====CodeCache====H-------T-------
+    //                |-------longest branch ---^
+    address longest_branch_start = (target < CodeCache::low_bound()) ? CodeCache::high_bound() - NativeInstruction::instruction_size
+                                                                     : CodeCache::low_bound();
+    need_trampoline = !reachable_from_branch_at(longest_branch_start, target);
```

Comment:
We have only +/-1Mb range in the riscv architecture, which is not enough for CodeCache in most case. In contrast, the branch range is 128 Mb in AArch64. So, is this modification necessary for the rv port?

Reply:
Hi @yadongw , thanks for reviewing.
Indeed in riscv64 CodeCache is basically not smaller than branch range (CodeCache initial size is usually 48M by default and the reserved size is usually 240M by default), in riscv64 this optimization seems to be not too meaningful, I apologize for my oversight and I will close this PR.
Thanks again.